packages/fast-components/src/slider-label/slider-label.ts

```diff
@@ -0,0 +1,58 @@
+import { attr, FastElement, observable } from "@microsoft/fast-element";
+import { FASTSlider } from "../slider";
+import { SliderOrientation } from "../slider/slider";
+
+export class SliderLabel extends FastElement {
+    @observable
+    public positionStyle: string;
+
+    @observable
+    public root: HTMLDivElement;
+
+    @attr
+    public max;
+
+    @attr
+    public position: string;
+
+    @attr
+    public label: string;
+
+    @attr({ attribute: "hide-mark", mode: "boolean" })
+    public hideMark: boolean = false;
+
+    private sliderMax: number;
+    private sliderMin: number;
+    private sliderOrientation: SliderOrientation;
+
+    public connectedCallback(): void {
+        super.connectedCallback();
+        this.sliderMax = (this.parentNode as FASTSlider).max;
+        this.sliderMin = (this.parentNode as FASTSlider).min;
+        this.sliderOrientation = (this.parentNode as FASTSlider).orientation;
+        this.positionStyle = this.positionAsStyle();
+        this.styleForOrientation();
+    }
+
+    private styleForOrientation = (): void => {
+        if (this.sliderOrientation === SliderOrientation.horizontal) {
+            this.root.classList.add("slider-label-horizontal");
+        } else {
+            this.root.classList.add("slider-label-vertical");
+        }
+    };
+
+    private positionAsStyle = (): any => {
+        const direction: string = (this.parentNode as FASTSlider).direction;
```

Comment:
is this guaranteed to have a direction property?

Reply:
modified to use interface and have defaults if parent does not implement said interface